rsa/key.py

```diff
@@ -570,81 +570,68 @@ def _save_pkcs1_pem(self):
         return rsa.pem.save_pem(der, b'RSA PRIVATE KEY')
 
 
-def find_p_q(nbits, getprime_func=rsa.prime.getprime, accurate=True):
+def find_p_q(nbits, getprime_func=rsa.prime.getprimebyrange, accurate="unused parameter retained for compatibility"):
     """Returns a tuple of two different primes of nbits bits each.
 
     The resulting p * q has exacty 2 * nbits bits, and the returned p and q
     will not be equal.
 
     :param nbits: the number of bits in each of p and q.
     :param getprime_func: the getprime function, defaults to
-        :py:func:`rsa.prime.getprime`.
+        :py:func:`rsa.prime.getprimebyrange`.
 
         *Introduced in Python-RSA 3.1*
 
-    :param accurate: whether to enable accurate mode or not.
     :returns: (p, q), where p > q
 
     >>> (p, q) = find_p_q(128)
     >>> from rsa import common
     >>> common.bit_size(p * q)
     256
-
-    When not in accurate mode, the number of bits can be slightly less
-
-    >>> (p, q) = find_p_q(128, accurate=False)
-    >>> from rsa import common
-    >>> common.bit_size(p * q) <= 256
-    True
-    >>> common.bit_size(p * q) > 240
-    True
-
     """
+    #constraints implemented are from FIPS 186-4 appendix B-3.1.2
+    #http://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.186-4.pdf#page=62
+
+    #primes are chosen to be between (2**nbits) and ceil(2**nbits/sqrt(2))
+    #this ensures that regardless of what primes are chosen the final modulus
+    #will be between (2**(2*nbits)) and (2**(2*nbits-1))
+
+    maximum=2**nbits#up to but not including
+    #multiply by fractional representation of 1/sqrt(2)(rounded up) for minimum
+    minimum=(maximum*0xb504f333f9de6484597d89b3754abea0)
```

Comment:
This really needs more explanation to explain why it is done this way (speed? precision? both? something else?). Furthermore, the code that is used to compute the constant needs to be callable and unit tested.

Also, avoid unnecessary parentheses.